### PR TITLE
Use version specifiers in PluginRequirement

### PR DESCRIPTION
**Pull request by mdellweg**
Review Checklist:
- [ ] An issue is properly linked. [feature and bugfix only]
- [ ] Tests are present or not feasible.
- [ ] Commits are split in a logical way (not historically).

### CODE REVIEW

**Comment by mdellweg** [comment]
When you realize, you reinvented the wheel...

**Comment by mdellweg** [comment]
`sed -i -E -e 's/PluginRequirement\("(.*)", (min=)?"([0-9.]*(dev)?)"(, inverted=True)?(, feature=.*)?\)/PluginRequirement("\1", specifier=">=\3"\5\6)/' pulpcore/cli/*/*.py pulp-glue/pulp_glue/*/*.py docs/*`
`sed -i -E -e 's/PluginRequirement\("(.*)", max="([0-9.]*(dev)?)"(, inverted=True)?(, feature=.*)?\)/PluginRequirement("\1", specifier="<\2"\4\5)/' pulpcore/cli/*/*.py pulp-glue/pulp_glue/*/*.py docs/*`

File: pulp-glue/pulp_glue/core/context.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ class PulpDomainContext(PulpEntityContext):
     ENTITIES = _("Pulp domains")
     HREF = "domain_href"
     ID_PREFIX = "domains"
-    NEEDS_PLUGINS = [PluginRequirement("core", "3.23.0.dev")]
+    NEEDS_PLUGINS = [PluginRequirement("core", specifier=">=3.23.0")]
```
Comment (ggainey) [inline pulp-glue/pulp_glue/core/context.py:91]: :+1:

File: pulpcore/cli/core/group.py [hunk 2]
```diff
@@ -77,7 +77,7 @@ def _object_required_grouprole_lookup_callback(
         "core:domain": PulpDomainContext,
     },
     "help": _("Domain the role is applied in"),
-    "needs_plugins": (PluginRequirement("core", "3.23.dev"),),
+    "needs_plugins": (PluginRequirement("core", specifier=">=3.23"),),
```
Comment (ggainey) [inline pulpcore/cli/core/group.py:80]: :+1:

File: pulpcore/cli/ansible/repository.py [hunk 3]
```diff
@@ -110,9 +110,7 @@ def repository(ctx: click.Context, pulp_ctx: PulpCLIContext, repo_type: str) ->
         "--gpgkey",
         callback=load_string_callback,
         needs_plugins=[
-            PluginRequirement(
-                "ansible", min="0.15.0.dev", feature="gpgkeys on ansible repositories"
-            )
+            PluginRequirement("ansible", min="0.15.0", feature="gpgkeys on ansible repositories")
```
Comment (ggainey) [inline pulpcore/cli/ansible/repository.py:113]:
```suggestion
            PluginRequirement("ansible", specifier=">=0.15.0", feature="gpgkeys on ansible repositories")
```

**Review by ggainey — CHANGES_REQUESTED** [review]

File: pulpcore/cli/ansible/repository.py (hunk 3 above)
Comment (mdellweg) [reply pulpcore/cli/ansible/repository.py:113]: programmers blind spot...

**Review by ggainey — APPROVED** [review]

**Pull request merged**